### ISSUE TITLE
Top level services should honor `"sharable":"singleton"` like lower level services do.

### ISSUE THREAD

**Issue by TheMosquito** [open-horizon/anax#843]
When a Service "X" is both a top level Service, and a direct or indirect dependency of some other top level Service "Y", then, even when Service X is defined as `"sharable":"singleton"` Anax will start 2 instances of Service X. This is counterintuitive, caused problems for a user (`@dcmartin`) today, and is likely to occur for other users once we go into in production.
The recommended fix is that even top level services should honor the `"sharable":"singleton"` attribute and only a single instance of Service X should be started in the above situation.
If you wish to follow the discussion in Slack see the #edge-fabric-dev channel's thread started by `@mosquito`  at 10:16AM Pacific time, Wednesday, Mar 27, 2019.

**Comment by dabooz** [comment]
I disagree with the assertion of how this should work. Top level services have their own lifecycle and do not need to honor the singleton attribute. I will leave this open for discussion.

**Comment by TheMosquito** [comment]
@dabooz Are you saying that the "sharable" attribute is completely ignored when a service happens to be included as a top level service? I think that is a problematic approach, and counter-intuitive from a user perspective. Often a service will be designated as singleton when a single instance is absolutely required for proper function. For example, a local MQTT broker, used by multiple other services. If multiple instances are launched then the resulting deployment will simply not work.

**Comment by bmpotter** [comment]
Dave,

I'm guessing one of the things you are concerned about is that currently each top-level service and its required services represent a network sandbox. If you implemented what glen is requesting, another network sandbox would have a network connection into 1 of the services in a different network sandbox. Are there other things you are worried about?

I agree with glen that 1 of the reasons to use `"sharable":"singleton"` is if a service needs an exclusive OS resource. If this is the reason, then when they add it as a top-level service this will be defeated.

And the usability issue is something we need to address. Currently if a service has `"sharable":"singleton"` set and they create a pattern with this service at the top level and required by another top level, we essentially ignore the setting w/o telling them. So if we decide not to support what glen is suggesting, we need to at least warn the user about this.

Wasn't the `agreementLess` attribute in the pattern created to address this exact problem? The swagger doc for the POST pattern says:
```
"agreementLess": false,  // only set to true if the same svc is both top level and required by another svc
```

Do we still support this? I think we added this for the wiotp mqtt broker, because it they needed it to always be there (so attached devices could use it), but also wanted edge services to be able to require it.

**Comment by dabooz** [comment]
yes, use agreementless for this use case

**Comment by dcmartin** [comment]
A service is a service, regardless of deployment as top-level or required; either the `singleton` directive is enforced for all services, or you break the design by defining a special case for top-level services which, in essence, defeats the purpose of singleton.
